Kernel/Devices/Audio/IntelHDA/Controller.h

```diff
@@ -26,7 +27,8 @@ class Codec;
 
 class Controller final
     : public AudioController
-    , public PCI::Device {
+    , public PCI::Device
+    , public IRQHandler {
```

Comment:
Hmm, is it possible that we could have a separate class to use `PCIIRQHandler` instead? does Intel HDA support MSI/MSI-X interrupts?

Reply:
Looks like a drop-in replacement of `IRQHandler` by `PCIIRQHandler` worked just fine! Based on my tests HDA supports MSI-X.